providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -205,12 +250,12 @@ public FlagdProvider setDeadline(long deadlineMs) {
     }
 
     @Override
-    public void setEventStreamAlive(boolean alive) {
+    public void setState(ProviderState state) {
```

Comment:
I don't think we need this method. In fact, I'm not sure why we aren't see an error in the C with the `@Override` annotation, since this doesn't actually override anything.

The state is something the provider should manage internally. It's only to signal to the SDK that the provider is ready (the interface only defines a getter). Basically you can just set a private backing field to READY/ERROR after init, and with disconnects, etc.

Reply:
Public modifiers comes from implementing an interface method of the [EventStreamCallback](https://github.com/open-feature/java-sdk-contrib/blob/main/providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/EventStreamCallback.java#L7)
I planned to work on this details as part of #365 since it requires quite some rework on the code

Reply:
Ya, I see that now. My local IDE somehow hand't caught up and was showing a compilation error here.